weaver/samples/corda/corda-simple-application/clients/src/main/kotlin/com/cordaSimpleApplication/client/InteropManager.kt

```diff
@@ -50,32 +54,39 @@ class RequestStateCommand : CliktCommand(help = "Requests state from a foreign n
                 password = "test",
                 rpcPort = config["CORDA_PORT"]!!.toInt())
         try {
+            val relayOptions = RelayOptions(
+                useTlsForRelay = config["RELAY_TLS"]!!.toBoolean(),
+                relayTlsTrustStorePath = config["RELAY_TLSCA_TRUST_STORE"]!!,
+                relayTlsTrustStorePassword = config["RELAY_TLSCA_TRUST_STORE_PASSWORD"]!!,
+                tlsCACertPathsForRelay = config["RELAY_TLSCA_CERT_PATHS"]!!
+            )
+            val k: String = if (key != null) {
+                key!!
+            } else {
+                "external_state"
+            }
+            val args: List<Any> = listOf<Any>(k, arrayOf<String>())
             InteroperableHelper.interopFlow(
-                rpc.proxy, 
-                localRelayAddress, 
-                externalStateAddress,
+                rpc.proxy,
+                arrayOf(externalStateAddress),
+                localRelayAddress,
                 networkName,
-                listOf<Party>(),
-                config["RELAY_TLS"]!!.toBoolean(),
-                config["RELAY_TLSCA_TRUST_STORE"]!!,
-                config["RELAY_TLSCA_TRUST_STORE_PASSWORD"]!!,
-                config["RELAY_TLSCA_CERT_PATHS"]!!
+                false,
+                "com.cordaSimpleApplication.flow.CreateFromExternalState",
+                args,
+                1,
+                externalStateParticipants = listOf<Party>(),
+                relayOptions = relayOptions
```

Comment:
This function call seems to be taking many more parameters than the function definition requires. Am I missing something? The "invoke object" seems to be decomposed into its components here rather than be a single parameter.